scripts/build-toolchains.sh

```diff
@@ -153,8 +152,19 @@ if [ "$FASTINSTALL" = "false" ]; then
         cd "$CHIPYARD_DIR/toolchains/$TOOLCHAIN"
         check_version automake 1.14 "OpenOCD build"
         check_version autoconf 2.64 "OpenOCD build"
-        build_project riscv-openocd --prefix=$RISCV --enable-remote-bitbang --enable-jtag_vpi --disable-werror
+        build_project riscv-openocd --prefix="${RISCV}" --enable-remote-bitbang --enable-jtag_vpi --disable-werror
         echo -e "\\nRISC-V OpenOCD installation completed!"
-        cd $RDIR
+        cd "$RDIR"
     fi
 fi
+
+cd "$RDIR"
+
+{
```

Comment:
The reason I had these before the EC2 check was in-case the ec2 check fails because of the some corner case, the sourceme file is still generated and the build essentially finished without a problem

Reply:
I made this change since I took `env.sh` being generated is a indication of success, which would not be true if OpenOCD was supposed to build but failed.  Not having `env.sh` is a clear sign to a user not necessarily paying attention to the logs that something went wrong.

I can see the value in having `env.sh` produced earlier since there's no way to resume without rebuilding everything entirely from scratch (which is another issue).  The commit has to be amend anyway since I just noticed `$DTCversion` is never defined.

Reply:
Yeah, I think OpenOCD is used in a very restricted set of use-cases (and most user will not need it), so indicating that the entire build process failed just because of OpenOCD would be any overkill. There is definitely a clear error message when OpenOCD fails